addons/stock/report/stock_forecasted.py

```diff
@@ -192,7 +192,10 @@ def _get_report_moves_fields(self):
 
     def _get_report_lines(self, product_template_ids, product_ids, wh_location_ids, wh_stock_location, read=True):
 
-        def _get_out_move_reserved_data(out, linked_moves, used_reserved_moves, currents):
+        def _get_out_move_reserved_data(out, linked_moves, used_reserved_moves, currents, wh_stock_location=wh_stock_location):
+            wh_stock_sub_location_ids = set(
+                (wh_stock_location.search([('id', 'child_of', wh_stock_location.id)]) - wh_stock_location)._ids
```

Comment:
As `_get_out_move_reserved_data` is called inside two nested `for` loops, this search will be done for each out move for each product. While always searching the same ids. I highly suggest you move that outside of the function. :smile: 

Especially since you're already doing that search outside. :eyes: